doc/guides/device_mgmt/index.rst

```diff
@@ -39,14 +39,466 @@ functionality over BLE and serial.
+
+   :option:`CONFIG_STAT_MGMT_MAX_NAME_LEN` sets the maximum length of a section
+   name that can can be accepted as parameter for showing the section data, and
+   might require tweaking for long section names.
+
+The final steps to use a statistics section is to initialize and register it::
+
+  rc = STATS_INIT_AND_REG(my_stats, STATS_SIZE_32, "my_stats");
+  assert (rc == 0);
+
+In the running code a statistics counter can be incremented by 1 using
+``STATS_INC``, by N using ``STATS_INCN`` or reset with ``STATS_CLEAR``.
+
+Let's suppose we want to increment those counters by ``1``, ``2`` and ``3``
+every second. To get a list of stats::
+
+  $ mcumgr --conn acm0 stat list
+  stat groups:
+    my_stats
+
+To get the current value of the counters in ``my_stats``::
+
+  $ mcumgr --conn acm0 stat my_stats
+  stat group: my_stats
+        13 my_stat_counter1
+        26 my_stat_counter2
+        39 my_stat_counter3
+
+  $ mcumgr --conn acm0 stat my_stats
+  stat group: my_stats
+        16 my_stat_counter1
+        32 my_stat_counter2
+        48 my_stat_counter3
+
+When :option:`CONFIG_STATS_NAMES` is disabled the output will look like this::
+
+  $ mcumgr --conn acm0 stat my_stats
+  stat group: my_stats
+         8 s0
+        16 s1
+        24 s2
+
+.. _fs_mgmt:
+
+Filesystem Management
+*********************
+
+The filesystem module is disabled by default because of security concerns,
+because of a lack of access control every file in the FS will be accessible,
+including secrets, etc. To enable it :option:`CONFIG_MCUMGR_CMD_FS_MGMT` must
+be set (``y``). Once enabled the following sub-commands can be used::
+
+  mcumgr <connection-options> fs download <remote-file> <local-file>
+  mcumgr <connection-options> fs upload <local-file> <remote-file>
+
+Using the ``fs`` command, requires :option:`CONFIG_FILE_SYSTEM` to be enabled,
+and that some particular filesystem is enabled and properly mounted by the running
+application, eg for littefs this would mean enabling
+:option:`CONFIG_FILE_SYSTEM_LITTLEFS`, defining a storage partition :ref:`flash_map_api`
+and mounting the filesystem in the startup (:c:func:`fs_mount`).
+
+Uploading a new file to a littlefs storage, mounted under ``/lfs`` can be done with::
+
+  $ mcumgr -c acm0 fs upload foo.txt /lfs/foo.txt
+  25
+  Done
+
+Where ``25`` is the size of the file.
+
+For downloading a file, let's first use the ``fs`` command
+(:option:`CONFIG_FILE_SYSTEM_SHELL` must be enabled) in a remote shell to create
+a new file::
+
+  uart:~$ fs write /lfs/bar.txt 41 42 43 44 31 32 33 34 0a
+  uart:~$ fs read /lfs/bar.txt
+  File size: 9
+  00000000  41 42 43 44 31 32 33 34 0A                       ABCD1234.
+
+Now it can be downloaded using::
+
+  $ mcumgr -c acm0 fs download /lfs/bar.txt bar.txt
+  0
+  9
+  Done
+  $ cat bar.txt
+  ABCD1234
+
+Where ``0`` is the return code, and ``9`` is the size of the file.
+
+.. warning::
+
+   The commands might exhaust the system workqueue if its size is not large
```

Comment:
```suggestion
   The commands might exhaust the system workqueue, if its size is not large
```

Reply:
I am not sure about this, but sentence without the `if` clause is also correct, so maybe the comma here is right.

Reply:
Done.